actix-web-codegen/src/route.rs

```diff
@@ -57,13 +59,15 @@ struct Args {
     path: syn::LitStr,
     guards: Vec<Ident>,
     wrappers: Vec<syn::Type>,
+    methods: Vec<GuardType>,
 }
 
 impl Args {
     fn new(args: AttributeArgs) -> syn::Result<Self> {
         let mut path = None;
         let mut guards = Vec::new();
         let mut wrappers = Vec::new();
+        let mut methods = Vec::new();
```

Comment:
we would register an extra redundant guard. This can be avoided by making the methods vec a set or returning a compile time error. _not sure which is the better approach_

Reply:
I would prefer using a set AND returning an error if method is met more than once. Luckily, `HashSet::insert` returns a bool indicating whether this method was already present inside.